src/analyses/base.ml

```diff
@@ -2627,13 +2651,13 @@ struct
     if get_bool "sem.noreturn.dead_code" && Cil.hasAttribute "noreturn" f.vattr then raise Deadcode else st
 
   let combine_st ctx (local_st : store) (fun_st : store) (tainted_lvs : AD.t) : store =
-    let ask = (Analyses.ask_of_ctx ctx) in
-    AD.fold (fun addr st ->
+    let ask = Analyses.ask_of_ctx ctx in
+    AD.fold (fun addr (st: store) ->
         match addr with
         | Addr.Addr (v,o) ->
           if CPA.mem v fun_st.cpa then
             let lval = Addr.Mval.to_cil (v,o) in
-            let address = eval_lv ask ctx.global st lval in
+            let address = eval_lv ~ctx lval in
```

Comment:
This is now semantically different, it used to evaluate in `st` which changes during the fold, but now it's evaluated in `ctx.local`. We'll have to check if this is still the right thing to do or not.

Reply:
Should I just add the `st` parameter back here and for the cases below as of now, or are we planning on looking into those more deeply?

Reply:
This can be bypassed entirely because some stupid conversions are happening here:
`addr` (`Addr.t`) → `(v, o)` (`Addr.Mval.t`) → `lval` (`lval`) → `address` (`AD.t`).
Instead this could be directly done with
```ocaml
let address = AD.singleton addr in
```